flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisITCase.java

```diff
@@ -143,8 +143,13 @@ private String stopWithSavepoint() throws Exception {
     }
 
     private static class WaitingMapper implements MapFunction<String, String> {
-        static CountDownLatch firstElement = new CountDownLatch(1);
-        static volatile boolean stopped = false;
+        static CountDownLatch firstElement;
+        static volatile boolean stopped;
+
+        WaitingMapper() {
```

Comment:
what was the problem with the previous code here?

Reply:
The previous version was not well-behaved if you rerun the test inside a loop (globals were not reset properly).

Reply:
makes sense, +1 then